src/sections/blog-posts.liquid

```diff
@@ -11,74 +11,90 @@
-    <div class="blog-post__item">
-      {% if article.image %}
+  {%- comment -%}
+    Due to accessibility requirement, the Blog entries list functions
+    as a list of content items. Non-visual users will expect lists of content
+    to appear in a list structure, allowing for easier understanding and navigation of list items.
+  {%- endcomment -%}
+  <ul class="blog-post">
+    {% for article in blog.articles limit: section.settings.post_limit %}
+      <li class="blog-post__item">
+        {%- comment -%}
+          Due to accessibility requirement, we are regrouping all logical elements into the <a> element.
+          If a screen reader user is tabbing through the page, or browsing by links on the page with duplicated links,
+          they will appear identical which may lead to a confusing user experience.
+        {%- endcomment -%}
         <a href="{{ article.url }}">
-          {% include 'responsive-image' with
-              image: article.image,
-              max_width: 400,
-              max_height: 400
-          %}
+          {% if article.image %}
+            {% include 'responsive-image' with
+                image: article.image,
+                max_width: 400,
+                max_height: 400
+            %}
+          {% endif %}
+          <h3>{{ article.title }}</h3>
         </a>
-      {% endif %}
 
-      <div>
-        <h2>
-          <a href="{{ article.url }}">{{ article.title }}</a>
-        </h2>
+        <div>
+          {% if section.settings.blog_show_author %}
+            <span>{{ 'blogs.article.by_author' | t: author: article.author }}</span>
+          {% endif %}
 
-        {% if section.settings.blog_show_author %}
-          <span>{{ 'blogs.article.by_author' | t: author: article.author }}</span>
-        {% endif %}
+          {% if section.settings.blog_show_date %}
+            <p>
+              {{ article.published_at | time_tag: format: 'month_day_year' }}
+            </p>
+          {% endif %}
 
-        {% if section.settings.blog_show_date %}
-          <p>
-            {{ article.published_at | time_tag: format: 'month_day_year' }}
-          </p>
-        {% endif %}
+          <div>
+            {% if article.excerpt.size > 0 %}
+              {{ article.excerpt }}
+            {% else %}
+              {{ article.content | strip_html | truncate: 150 }}
+            {% endif %}
+          </div>
 
-        <div>
-          {% if article.excerpt.size > 0 %}
-            {{ article.excerpt }}
-          {% else %}
-            {{ article.content | strip_html | truncate: 150 }}
+          {% if article.tags.size > 0 %}
+            {%- comment -%}
+              The Tag link list functions as a list of content items.
+              Non-visual users will expect lists of content to appear
+              in a list structure, allowing for easier understanding and navigation of list items.
+            {%- endcomment -%}
+            <ul>
```

Comment:
Can you add an `aria-label="tags"` here so when we navigate to the list we get some context ?

Reply:
Good point, thanks!